lib/model/services/ble/machine_service.dart

```diff
@@ -916,9 +916,8 @@ class EspressoMachineService extends ChangeNotifier {
       var cs = Shot();
       cs.coffee.targetId = coffeeService.selectedCoffeeId;
       cs.recipe.targetId = coffeeService.selectedRecipeId;
-      var save = settingsService.savePrePouring;
       cs.shotstates.addAll(
-          shotList.entries.where((element) => (element.isPouring == true || save) && element.isInterpolated == false));
+          shotList.entries.where((element) => element.isInterpolated == false));
```

Comment:
No additional filtering is required here, since the heating phase gets wiped out in the main loop if the user hasn't opted in to saving it.